Style See the Fishway tab

alexelash opened the pull request:
## Overview
A style pass of See the Fishway.

Connects #93

### Demo
<img width="1792" alt="Screen Shot 2019-07-05 at 2 01 54 PM" src="https://user-images.githubusercontent.com/5672295/60739035-5c751600-9f2e-11e9-8df3-d9f7e2ad0619.png">


### Notes
- There's an error that I'm seeing on `develop` as well as this branch... but not on staging: 
```
Warning: Failed prop type: The prop `fish.path` is marked as required in `Sidebar`, but its value is `undefined`.
```
- The scrubber was set to replay when the user moved the track handle. This pattern felt strange to me, so I removed the code that did this. When the user interacts with the scrubber, it pauses and stays paused until they hit the play button. 
- It looks like there are more videos here than in the current app (which has 9)—
<img width="878" alt="Screen Shot 2019-07-05 at 1 59 58 PM" src="https://user-images.githubusercontent.com/5672295/60738692-356a1480-9f2d-11e9-8c52-9404f4d1b40e.png"> If we can remove some of them, that would be great so that they can all fit comfortably on the screen. I would then alter the code slightly to accommodate.
- It would be nice if the latin name for each "fish" weren't required, so that we could hide the placeholder shown here entirely, but I'm not sure how to do this. Should I enter another issue? It seems like a detail that we could save for the end of the project: <img width="614" alt="Screen Shot 2019-07-05 at 2 05 53 PM" src="https://user-images.githubusercontent.com/5672295/60738996-394a6680-9f2e-11e9-902b-ed72da33b14b.png">
- I _think_ I can't style the live version yet until we do #74. Created #108 to cover this work.


## Testing Instructions
* git pull
* Go to See the Fishway tab
* Select different videos by their button
* Play with scrubber, note that the video stops playing once the scrubber is interacted with and doesn't play again without the user clicking the play button.

Commit:
Style See the Fishway

fungjj92 commented:
Nice work! I tested on the Touchscreen and all the videos fit well, actually. There's even 1 square in the video grid left unfilled, which will accommodate the "live" video button without elongating the page content. We could remove some options though, assuming video parity is ok.

- re: video scrubber. Your change makes sense! The scrubber works well on my laptop, but will need a little work for touchscreen compatibility. The touch range is quite limited/precise (both the slider and play/pause buttons) and there's a lag when dragging the play position. Of these observations, what can be done in this PR I think is making those controls bigger. What do you think?
- re: console error. News to me! You can remove line 7 entirely in `HighlightFish.jsx` https://github.com/azavea/pwd-fishway/blob/a598e72282a59536d1cfcd3ae3d57005b86ffd59/src/app/src/util/HighlightFish.js#L7
- re: scientific subtitle: We can make scientific name optional

Update: For bullets 2 & 3, I've made changes accordingly in PR #111

alexelash commented:
@fungjj92 Great! Maybe it's a question for Victoria—I'm curious if there's a reason why they don't have those videos (maybe they didn't get approval for some reason or other).

- Cool, I'll make some adjustments to the scrubber and test on the actual machine.
- Sounds good, will do.
- Making it optional would be perfect.

Commit:
Make scrubber handle grab area bigger

Commit:
Prevent video squares from stretching

Commit:
Add styles for Live feed

alexelash commented:
@fungjj92 Some of the changes in the most recent commits:
- The scrubber handle is visually the same size but has a larger touch area. You can test by clicking slightly to the right or left of the scrubber itself.
- The video squares were stretching strangely when I played with the screen size, so this is now fixed.
- Now that the live feed is in the app, I added necessary styles:

<img width="1792" alt="Screen Shot 2019-07-15 at 12 59 32 PM" src="https://user-images.githubusercontent.com/5672295/61234181-6a7e2000-a700-11e9-96b0-d95661148cb0.png">

One pretty big issue is that the live video we're using isn't a perfect square, so the thumbnail feels wrong. Setting the width of the `img` through the width and height props on the JSX doesn't change the actual image proportions. 

It looked like the _actual_ live video proportions will eventually be square (based on the previous app). If it doesn't end up being that way, we should plan to handle the thumbnail differently (perhaps as a static image or without an image at all).

fungjj92 commented on src/app/src/components/VideoCard.js:
```diff
@@ -1,23 +1,72 @@
-            selected={selected}
-        />
+        <StyledBox>
+            {(() => {
```
Curious about this block of code being a function? Unless I'm misunderstanding intent, we could write more simply as...
```
...
const liveVideoHeading = (...html here...);
return (
...
{fish.isLiveFeed && liveVideoHeading}
<StyledVideoCard ... />
...
);
```

alexelash replied:
🙇‍♀Oops, nope, you got it! That's much better. Updated in my most recent commit to work this way.

fungjj92 commented on src/app/src/util/constants.js:
```diff
@@ -213,6 +213,7 @@ export const LIVE_FEED_MOCK_FISH = {
     timestamp: new Date().getTime(),
     video: LIVE_FEED_URL,
     photo: LIVE_FEED_URL,
+    isLiveFeed: true,
```
We detect if the fish is the live feed fish two different ways. I like this field better than a name check (which is what is done here https://github.com/azavea/pwd-fishway/pull/109/files#diff-11b2b3c1e45b70b30d73055c028f1a06R75) and we'll want to propogate that accordingly -- in `HighlightFish.js` add a field `isLiveFeed: bool` and import the `bool` prop-type, but don't make it required.

Update https://github.com/azavea/pwd-fishway/pull/109/files#diff-11b2b3c1e45b70b30d73055c028f1a06R75 to use that field

fungjj92 commented on src/app/src/components/VideoCard.js:
```diff
@@ -1,23 +1,72 @@
-        />
+        <StyledBox>
+            {(() => {
+                if (fish.isLiveFeed === true)
```
the condition can just be `(fish.isLiveFeed)`

fungjj92 reviewed (changes requested):
Thanks for incorporating my last comments and rebasing! I tested the slider on the touchscreen and the increased touch range feels better. Looking good!!

OK @ revisiting how to address the live feed dimensions, when we get it!

Commit:
Use isLiveFeed bool for sidebar and export prop correctly

Commit:
Improve if statement for live feed video heading

alexelash commented:
@fungjj92 Thanks for the review! Ready for another look.

fungjj92 reviewed (approved):
🥇
